swdb_2018_tools/ephys/get_stimulus_blocks.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+
+"""
+Created on Mon Aug 27
+
+@author: shawno
+"""
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+
+def get_stimulus_blocks(data_set):
+    '''Return dataframe with start and end times for each stimulus block '''
+    stim_blocks = []
+    for i,k in enumerate(data_set.stim_tables.keys()):
+
+        stim_table = data_set.stim_tables[k]
+        s = stim_table.start.values
+        s_end = stim_table.end.values
+        idx = np.where(np.diff(s)>100)[0]
+
+        if len(idx)==0:
+            stim_blocks.append((k,s[0],s_end[-1]))
+
+        elif len(idx)==1:
+            stim_blocks.append((k,s[0],s_end[idx[0]-1]))
+            stim_blocks.append((k,s[idx[0]+1],s_end[-1]))
+
+        elif len(idx)==2:
+            stim_blocks.append((k,s[0],s_end[idx[0]-1]))
+            stim_blocks.append((k,s[idx[0]+1],s_end[idx[1]-1]))    
+            stim_blocks.append((k,s[idx[1]+1],s_end[-1]))  
+    
+    stim_blocks = pd.DataFrame(stim_blocks,columns=['stimulus_type','start','end'])
+            
+    return stim_blocks
+
+def plot_stimulus_blocks(data_set,ax=[]):
```

Comment:
watch out for assigning mutable objects as default parameter values - not an issue here, but unsafe.

Reply:
maybe `None` and a check like `if ax is None:`